artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnectionManager.java

```diff
@@ -0,0 +1,242 @@
+import java.util.concurrent.Executor;
+import java.util.concurrent.locks.Lock;
+
+import io.netty.channel.ChannelPipeline;
+import org.apache.activemq.artemis.api.core.ActiveMQException;
+import org.apache.activemq.artemis.api.core.Interceptor;
+import org.apache.activemq.artemis.api.core.client.ClientSessionFactory;
+import org.apache.activemq.artemis.core.config.amqpBrokerConnectivity.AMQPBrokerConnectConfiguration;
+import org.apache.activemq.artemis.core.remoting.impl.netty.NettyConnection;
+import org.apache.activemq.artemis.core.remoting.impl.netty.NettyConnector;
+import org.apache.activemq.artemis.core.remoting.impl.netty.NettyConnectorFactory;
+import org.apache.activemq.artemis.core.server.ActiveMQComponent;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.protocol.amqp.broker.ProtonProtocolManager;
+import org.apache.activemq.artemis.protocol.amqp.broker.ProtonProtocolManagerFactory;
+import org.apache.activemq.artemis.spi.core.protocol.RemotingConnection;
+import org.apache.activemq.artemis.spi.core.remoting.ClientConnectionLifeCycleListener;
+import org.apache.activemq.artemis.spi.core.remoting.ClientProtocolManager;
+import org.apache.activemq.artemis.spi.core.remoting.Connection;
+import org.apache.activemq.artemis.spi.core.remoting.SessionContext;
+import org.apache.activemq.artemis.spi.core.remoting.TopologyResponseHandler;
+
+public class AMQPBrokerConnectionManager implements ActiveMQComponent, ClientConnectionLifeCycleListener {
+   private final ProtonProtocolManagerFactory protonProtocolManagerFactory;
+   private final ActiveMQServer server;
+   private volatile boolean started = false;
+
+   List<AMQPBrokerConnectConfiguration> amqpConnectionsConfig;
+   List<AMQPBrokerConnection> amqpOutgoingConnections;
+   ProtonProtocolManager protonProtocolManager;
+
+   public AMQPBrokerConnectionManager(ProtonProtocolManagerFactory factory, List<AMQPBrokerConnectConfiguration> amqpConnectionsConfig, ActiveMQServer server) {
+      this.amqpConnectionsConfig = amqpConnectionsConfig;
+      this.server = server;
+      this.protonProtocolManagerFactory = factory;
+   }
+
+   @Override
+   public void start() throws Exception {
+      if (!started) {
+         started = true;
+      }
+
+      NettyConnectorFactory factory = new NettyConnectorFactory().setServerConnector(true);
+      protonProtocolManager = (ProtonProtocolManager)protonProtocolManagerFactory.createProtocolManager(server, null, null, null);
+      NettyConnector bridgesConnector = (NettyConnector)factory.createConnector(null, null, this, server.getExecutorFactory().getExecutor(), server.getThreadPool(), server.getScheduledPool(), new ClientProtocolManagerWithAMQP(protonProtocolManager));
+      bridgesConnector.start();
+
+      amqpOutgoingConnections = new ArrayList<>();
+
+
+      for (AMQPBrokerConnectConfiguration config : amqpConnectionsConfig) {
+         System.out.println("Connecting " + config);
```

Comment:
System.out.println still needs updated (or perhaps comment added to indicate its deliberately not using a logger)

Reply:
I did not mean System.out... I thought I had removed it...


I guess I did some gif-fu push -f that I lost it.. will fix it.